src/plugins/azgenerator/CodeModelAzImpl.ts

```diff
@@ -162,6 +177,123 @@ export class CodeModelCliImpl implements CodeModelAz {
+                                        let names = this.Method_Name.split(' ');
+                                        if (flattenedNames) {
+                                            flattenedLen = flattenedNames.length;
+                                            for(let item of flattenedNames) {
+                                                mapName.push(ToSnakeCase(item));
+                                            }
+                                            if (mapName[0] == 'properties' || mapName[0] == 'parameters') {
+                                                delete mapName[0];
+                                            } else if (names.length > 1 && mapName[flattenedLen - 1].toLowerCase() == names[0].replace(/-/g, '')) {
+                                                delete mapName[0];
+                                            }
+                                            if(mapName.length > 0) {
+                                                paramFlattenedName = mapName.join("_");
+                                            }
+                                        }
+                                        if (nameParamReference.has(paramFlattenedName) && nameParamReference.get(paramName).schema != param.schema) {
+                                            let preParam = nameParamReference.get(paramFlattenedName);
+                                            let preMapName: Array<string> = [];
+                                            let preFlattenedNames = preParam?.['targetProperty']?.['flattenedNames'];
+                                            let preParamFlattenedName = preParam.language['az'].mapsto;
+                                            let preTmpName = preParamFlattenedName;
+                                            if (preFlattenedNames) {
+                                                for(let item of preFlattenedNames) {
+                                                    preMapName.push(ToSnakeCase(item));
+                                                }
+                                                if (preMapName.length > 0) {
+                                                    preTmpName = preMapName.join('_');
+                                                }
+                                            }
+                                            if (flattenedNames) {
+                                                mapName = [];
+                                                for(let item of flattenedNames) {
+                                                    mapName.push(ToSnakeCase(item));
+                                                }
+                                                if(mapName.length > 0) {
+                                                    tmpName = mapName.join("_");
+                                                }
+                                            }
```

Comment:
is this what you want?
if(flattenedNames && flattendNames.length>0) 
  tmpName = flattenedNames.map(n => ToSnakeCase(n)).join("_")

Reply:
oh  yeah

Reply:
Actually we need a tmp array for this case. you won't be able to add the removed item back when collision is found if we remove element directly in flattenedNames. the other cases has been changed.